src/stream/src/executor/merge.rs

```diff
@@ -369,17 +467,11 @@ impl Stream for SelectReceivers {
         assert!(self.active.is_terminated());
         let barrier = self.barrier.take().unwrap();
 
-        // If this barrier asks the actor to stop, we do not reset the active upstreams so that the
-        // next call would return `Poll::Ready(None)` due to `is_terminated`.
         let upstreams = std::mem::take(&mut self.blocked);
-        if barrier.is_stop(self.actor_id) {
```

Comment:
I think it's safe to ignore the `is_stop` in the `SelectReceiver`, because the merge executor stops when seeing the stop barrier, and should not poll the `SelectReceiver` later.